### DISCUSSION TITLE
Alternate Main Display screen with Full Weather

### DISCUSSION THREAD

**Discussion by darianbjohnson** [Phambili-Tech/Newt_Display#11] [General]
I got a request to provide additional weather details. I'm still fooling around with formatting... but wanted to share my thinking. Welcome any thoughts and feedback.

I'd build this as a separate screen and would give users to option to make it the main screen.

![image](https://user-images.githubusercontent.com/21129686/181140686-6d8dd6b6-35ff-4265-a93b-f1f935438c24.png)

**Comment by darianbjohnson** [comment]
Here's another view (dimensions are a bit off

![Full_Weather_Screen](https://user-images.githubusercontent.com/21129686/181140754-9d7030b2-a28f-4aba-9fcf-e7a3a5eb6be6.png)
)

**Comment by andypiper** [comment]
I like the idea of shrinking the date and time to make more room for weather information. Good thoughts here so far.

**Comment by darianbjohnson** [comment]
The icons were a little "funky", so I went with text. Still draft (values are hard coded: need a proper degree sign for temp, 12/24 setting needed, etc.). Also sticking with the "large" time format for now (it's easier to alter the main screen to either call a quote render or the new "currentWeatherDetailsRender") function. 

![image](https://user-images.githubusercontent.com/21129686/183560884-20f600f3-d7fa-43fb-99c5-497bc68db0c9.png)

Right now I'm using current weather values from the "current" API call from [openweathermap ](https://openweathermap.org/price)- which only provides current temp, feels like temp, humidity, visibility, and wind speed. The "One API" call provides way more information, but that cost 0.015 for each call (so that's out of the question for the "freeware/opensource" version of the code).

Not sure if visibility is really a "value-add" measurement to display. Maybe I should switch with either air pressure (also from the openweathermap api), air quality (which I get from a different free-to-use API), or UV - which is again included in the "one call" api - but maybe I can get from another source.

Welcome any thoughts on other weather APIs that I should use. Newt makes a "get current weather" call every 30 mins (might shorten to 20 mins). Assuming I stick to two calls an hour ....
1 Newt over 1 day: 
- one 5-day/24-hour forecast call, 
- forty-eight current weather calls
- (round to 50 for easy math)

Therefore:
50 calls/day * 31 days/month * 300 units = 465,000 calls a month.

A combination of the "free tier" (called twice/hour for current and 3-hour forecasts) and the One Call API (for called once a day for 5-day forecast) allows me to serve all Newt customers at the lowest cost.

Thoughts?

**Reply by darianbjohnson** [reply]
Hmmm - I just found [https://www.weatherapi.com/docs/](https://www.weatherapi.com/docs/) - their prices are reasonable ($4 or $7 a month - less than the price of a chick-fil-a meal). And they provide a bunch of information. Definitely worth exploring

Note: My goal is to spend less than $15 a month providing web/information services for Newt... so this would fit into the budget (I already spend $7/month on a cloud MQTT - though I am thinking of standing up my own in the cloud for about $3.5/month).

**Reply by andypiper** [reply]
As a general thought on the display - I like it, and the additional information that is provided. I'd probably suggest air quality is more interesting than visibility for a lot of people / for me, but that's going to vary by user I expect.

**Reply by koocotte** [reply]
It's nice to see that 12/24 setting is coming, because nobody use 12h display here in France.

**Comment by tjfoth** [comment]
I REALLY love this!  This is what I was hoping for. I think Andy has a
great point about air quality.

I'd certainly like this to be the default screen.

One thing to consider is the quality of the data.  Again, if you allowed
people to provide their own keys, then there is no cost to you.  In fact,
you could actually take their keys and fetch it for them on AWS.

But this is looking great!

---
Tom Foth
***@***.***
***@***.***

**Comment by darianbjohnson** [comment]
Slight tweaks… added air quality and UV index
![34D9AAE5-491F-4A36-8CB2-F89FACC5AFE6](https://user-images.githubusercontent.com/21129686/183801584-ad023033-81b7-40f2-a1a0-34cab30b4c40.jpeg)

**Comment by darianbjohnson** [comment]
Slowly getting there…
![AFA82AF0-AF98-493E-99C1-C0B06E43BD8E](https://user-images.githubusercontent.com/21129686/186072722-6551ae6a-faf1-4ef5-8ee8-1a1d0bc5ab98.jpeg)
![DBDFE1D5-6224-403C-A5FF-6D361F138177](https://user-images.githubusercontent.com/21129686/186072737-13128d59-4a1c-44fb-b8f8-b1b021a72aa9.jpeg)

**Comment by tjfoth** [comment]
I'm happy to beta test this...😁😁😁

---
Tom Foth
***@***.***
***@***.***

**Reply by darianbjohnson** [reply]
I think having a handful of beta testers will be a great idea…. Code is still a few updates away from MVP… I’ll keep you posted

**Comment by darianbjohnson** [comment]
Finally have a version that I think looks good… now onto updating the other weather screens 
![B0931D41-8FAD-4A48-B431-8267CD7FBA7E](https://user-images.githubusercontent.com/21129686/187046126-0dace5cc-4f85-422a-aeb4-75048b045f3d.jpeg)

**Comment by tjfoth** [comment]
I'm all in on being a beta tester!

---
Tom Foth
***@***.***
***@***.***

**Comment by badslug** [comment]
I really like the weather centric main screen. One suggestion; I find the weekly forecast to be super handy (and sunset/sunrise not too useful). Could there either be another main screen option with the weekly forecast - basically take out the day forecast in the bottom right corner, move the humidity and uv index up to the same "row" as "feels like" and "air quality" and put the weekly forecast across the bottom row?

**Comment by darianbjohnson** [comment]
I've been thinking about changing the "Day" box... maybe instead using it for a weekly or hourly forecast instead... maybe I'll have two flavors (one with hourly, one with daily/weekly) that rotate every 5 mins....

**Reply by badslug** [reply]
That could work. Can it fit in that small box?

I'm actually not a huge fan of rotating items just because inevitably I'll want to see whatever is not shown and then have to wait for the interval to see that thing... Could we maybe do weekly numbers and a spark graph for the hourly?

I love the visualizations from the Weatherstrip app https://www.weatherstrip.app as a potential inspiration for the spark graph...

**Comment by darianbjohnson** [comment]
Not sure if it can fit into the box (and be legible)... I think you're probably right about giving users an option to choose from a number of different "home" screens. I'll also take a look at the weatherstrip app page for some ideas

**Comment by tjfoth** [comment]
Looking great!

Will one of the buttons take you to the hourly and daily forecasts?

---
Tom Foth
***@***.***
***@***.***
